# HG changeset patch
# User Rik <rik@octave.org>
# Date 1371930232 25200
#      Sat Jun 22 12:43:52 2013 -0700
# Node ID 62fa28af28428a09d1f60ca70836574c2f1c8de3
# Parent  84505f200e05c56e1cc16dd51656a1d4d643db64
doc: add citation() to documentation.

* doc/interpreter/octave.texi: Add new menu entry in Preface chapter.
* doc/interpreter/preface.txi: Add new node "Citing Octave in Publications".
* scripts/miscellaneous/citation.m: Improve docstring.

diff --git a/doc/interpreter/octave.texi b/doc/interpreter/octave.texi
--- a/doc/interpreter/octave.texi
+++ b/doc/interpreter/octave.texi
@@ -195,16 +195,17 @@ It corresponds to GNU Octave version @va
 * Operator Index::              An item for each documented operator.
 
 @detailmenu
  --- The Detailed Node Listing ---
 
 Preface
 
 * Acknowledgements::            
+* Citing Octave in Publications::
 * How You Can Contribute to Octave::  
 * Distribution::                
 
 Introduction
 
 * Running Octave::              
 * Simple Examples::             
 * Conventions::                 
diff --git a/doc/interpreter/preface.txi b/doc/interpreter/preface.txi
--- a/doc/interpreter/preface.txi
+++ b/doc/interpreter/preface.txi
@@ -59,16 +59,17 @@ easily.
 Everyone is encouraged to share this software with others under the
 terms of the GNU General Public License (@pxref{Copying}).  You are 
 also encouraged to help make Octave more useful by writing and 
 contributing additional functions for it, and by reporting any problems
 you may have.
 
 @menu
 * Acknowledgements::            
+* Citing Octave in Publications::
 * How You Can Contribute to Octave::  
 * Distribution::                
 @end menu
 
 @node Acknowledgements
 @unnumberedsec Acknowledgements
 @cindex acknowledgements
 @end ifclear
@@ -141,16 +142,30 @@ Department of Chemical and Biological En
 
 @item
 Richard Stallman, for writing GNU.
 @end itemize
 
 This project would not have been possible without the GNU software used
 in and to produce Octave.
 
+@node Citing Octave in Publications
+@unnumberedsec Citing Octave in Publications
+@cindex Citing Octave
+@cindex Citations
+
+In view of the many contributions made by numerous developers over many years
+it is common courtesy to cite Octave in publications when it has been used 
+during the course of research or the preparation of figures.  The
+@code{citation} function can automatically generate a recommended citation
+text for Octave or any of its packages.  See the help text below on how to
+use @code{citation}.
+
+@DOCSTRING(citation)
+
 @node How You Can Contribute to Octave
 @unnumberedsec How You Can Contribute to Octave
 @cindex contributing to Octave
 @cindex funding Octave development
 
 There are a number of ways that you can contribute to help make Octave a
 better system.  Perhaps the most important way to contribute is to write
 high-quality code for solving new problems, and to make your code freely
diff --git a/scripts/miscellaneous/citation.m b/scripts/miscellaneous/citation.m
--- a/scripts/miscellaneous/citation.m
+++ b/scripts/miscellaneous/citation.m
@@ -14,22 +14,22 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {Command} {} citation
 ## @deftypefnx {Command} {} citation @var{package}
-## Display instructions to cite GNU Octave and packages in publications.
+## Display instructions for citing GNU Octave or its packages in publications.
 ##
-## When called without an argument, displays information on how to cite the core
-## GNU Octave system in publications.  When given a package name @var{package},
-## display information on to cite it.  However, note that some packages may not
-## have instructions on how to cite them.
+## When called without an argument, display information on how to cite the core
+## GNU Octave system.  When given a package name @var{package}, display
+## information on citing the specific named package.  Note that some packages
+## may not yet have instructions on how to cite them.
 ##
 ## The GNU Octave developers and its active community of package authors have
 ## invested a lot of time and effort in creating GNU Octave as it is today.
 ## Please give credit where credit is due and cite GNU Octave and its packages
 ## when you use them.
 ##
 ## @end deftypefn
 
@@ -41,11 +41,14 @@ function citation (package = "octave")
   if (nargin > 1)
     print_usage ();
   else
     display_info_file ("citation", package, "CITATION");
   endif
 
 endfunction
 
+
+%% Test input validation
 %!error citation (1, 2)
 %!error <citation: PACKAGE must be a string> citation (1)
 %!error <citation: package .* is not installed> citation ("__NOT_A_VALID_PKG_NAME__")
+
